### ISSUE TITLE
support for laravel 7

### ISSUE THREAD

**Issue by cirolosapio** [dwightwatson/aggregate#6]

**Comment by cirolosapio** [comment]
i've tried to install but it fails
```
 composer require watson/aggregate
Using version ^3.0 for watson/aggregate
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for watson/aggregate ^3.0 -> satisfiable by watson/aggregate[3.0.0].
    - Conclusion: remove laravel/framework v7.0.8
    - Conclusion: don't install laravel/framework v7.0.8
    - watson/aggregate 3.0.0 requires illuminate/database ~6.0 -> satisfiable by laravel/framework[6.x-dev], illuminate/database[6.x-dev, v6.0.0, v6.0.1, v6.0.2, v6.0.3, v6.0.4, v6.1.0, v6.10.0, v6.11.0, v6.12.0, v6.13.0, v6.13.1, v6.14.0, v6.15.0, v6.15.1, v6.16.0, v6.17.0, v6.17.1, v6.18.0, v6.2.0, v6.3.0, v6.4.1, v6.5.0, v6.5.1, v6.5.2, v6.6.0, v6.6.1, v6.6.2, v6.7.0, v6.8.0].
    - Can only install one of: laravel/framework[6.x-dev, v7.0.8].
    - don't install illuminate/database 6.x-dev|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.0.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.0.1|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.0.2|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.0.3|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.0.4|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.1.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.10.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.11.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.12.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.13.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.13.1|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.14.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.15.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.15.1|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.16.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.17.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.17.1|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.18.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.2.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.3.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.4.1|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.5.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.5.1|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.5.2|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.6.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.6.1|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.6.2|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.7.0|don't install laravel/framework v7.0.8
    - don't install illuminate/database v6.8.0|don't install laravel/framework v7.0.8
    - Installation request for laravel/framework (locked at v7.0.8, required as ^7.0) -> satisfiable by laravel/framework[v7.0.8].


Installation failed, reverting ./composer.json to its original content.
```

any thoughts?

**Comment by dwightwatson** [comment]
Looks like I updated it incorrectly. I've yanked and re-published that tag. Sorry about that.

**Comment by cirolosapio** [comment]
Hi, i've tried to install but composer shows me:

```
composer require watson/aggregate

  [Composer\Downloader\TransportException]
  Content-Length mismatch, received 505177 bytes out of the expected 749758
```

After a search I realized that modifying the "config.json" to force https everything is fine:
```
composer config --list --global          // this will get the composer home path.
[home] /root/.composer                   // it's my composer home path.
```

I changed it to: 
```
{
  "config": {
    "github-protocols": [
      "https"
    ]
  },
  "repositories": {
    "packagist.org": {
      "type": "composer",
      "url": "https://packagist.org"
    }
  }
}
```

My question is: is it a problem with my composer configuration or it is a problem with your package?

Hope it helps

**Comment by dwightwatson** [comment]
I was able to install it on a new Laravel 7 project without issue. After a quick Google I came across the same solution as you that basically appears to be an issue with your Composer configuration, rather than anything on my end. Especially as other people have had the same issue installing Laravel which definitely doesn't have a problem. Sounds like forcing HTTPS is a good idea anyway, so if this resolved the problem for you then that's good news.
